analyzers/dataframe/src/ReconstructedParticleSource.cc

```diff
@@ -0,0 +1,300 @@
+    const edm4hep::RecoMCParticleLinkCollection &inLinkColl) {
+  edm4hep::ReconstructedParticleCollection result;
+  result.setSubsetCollection();
+
+  for (const auto &link : inLinkColl) {
+    const auto &mcParticle = link.getTo();
+    const auto &recoParticle = link.getFrom();
+    if (mcParticle.getGeneratorStatus() == m_status) {
+      result.push_back(link.getFrom());
+    }
+  }
+
+  return result;
+}
+
+// --------------------------------------------------------------------------
+ROOT::VecOps::RVec<float>
+getP(const edm4hep::ReconstructedParticleCollection &inColl) {
+  ROOT::VecOps::RVec<float> result;
+
+  for (const auto &particle : inColl) {
+    auto lVec = edm4hep::utils::p4(particle, edm4hep::utils::UseMass);
+    result.push_back(static_cast<float>(lVec.P()));
+  }
+  return result;
+}
+
+// --------------------------------------------------------------------------
+ROOT::VecOps::RVec<float>
+getPt(const edm4hep::ReconstructedParticleCollection &inColl) {
+  ROOT::VecOps::RVec<float> result;
+  result.reserve(inColl.size());
+
+  std::transform(
+      inColl.begin(), inColl.end(), std::back_inserter(result),
+      [](edm4hep::ReconstructedParticle p) { return edm4hep::utils::pt(p); });
+
+  return result;
+}
+
+// --------------------------------------------------------------------------
+ROOT::VecOps::RVec<float>
+getY(const edm4hep::ReconstructedParticleCollection &inColl) {
+  ROOT::VecOps::RVec<float> result;
+
+  for (const auto &particle : inColl) {
+    auto lVec = edm4hep::utils::p4(particle, edm4hep::utils::UseMass);
+    result.push_back(static_cast<float>(lVec.Rapidity()));
+  }
+
+  return result;
+}
+
+// --------------------------------------------------------------------------
+ROOT::VecOps::RVec<float>
+getE(const edm4hep::ReconstructedParticleCollection &inColl) {
+  ROOT::VecOps::RVec<float> result;
+
+  std::transform(
+      inColl.begin(), inColl.end(), std::back_inserter(result),
+      [](edm4hep::ReconstructedParticle p) { return p.getEnergy(); });
+
+  return result;
+}
+
+// --------------------------------------------------------------------------
+ROOT::VecOps::RVec<float>
+getMass(const edm4hep::ReconstructedParticleCollection &inColl) {
+  ROOT::VecOps::RVec<float> result;
+
+  std::transform(inColl.begin(), inColl.end(), std::back_inserter(result),
+                 [](edm4hep::ReconstructedParticle p) { return p.getMass(); });
+
+  return result;
+}
+
+// --------------------------------------------------------------------------
+ROOT::VecOps::RVec<float>
+getCharge(const edm4hep::ReconstructedParticleCollection &inColl) {
+  ROOT::VecOps::RVec<float> result;
+
+  std::transform(
+      inColl.begin(), inColl.end(), std::back_inserter(result),
+      [](edm4hep::ReconstructedParticle p) { return p.getCharge(); });
+
+  return result;
+}
+
+// --------------------------------------------------------------------------
+edm4hep::ReconstructedParticleCollection
```

Comment:
This could potentially return a `std::vector` or an `RVec`, because then we would save the creation of an additional collection at the end. However, I assume that would then make it hard to combine with other functions that expect a `ReconstructedParticleCollection`?

Reply:
The goal was to have ReconstructedParticleCollection as output. Maybe we can have both, but I would do this in the subsequet PR.

Is there any technical limitation in making EDM4hep collections sortable?

Reply:
There is no really easy way to make EDM4hep collections sortable, because the position in the collection is part of the ID of each object. However, we need to think about a way of making some of the operations on vectors / regular c++ collections possible also for EDM4hep / podio generated collections.